### ISSUE TITLE
Not all textboxes have a vertical resize

### ISSUE THREAD

**Issue by sofiamatulis** [Shopify/dawn#1151]
**Describe the current behavior**
Currently we have a vertical resize for the texbox in the cart note

**Describe the expected behavior**
We should have the vertical resize for all other textboxes (contact form, blog comment...)

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 1.0.0
- Chrome Version 91.0.4472.114
- macOS Version 11.3.1

**Possible solution**
From @svinkle : "If we're able to keep resizing and only show the visual affordance on mouse hover that'd be a good solution."

**Additional context/screenshots**
Original comment:
https://github.com/Shopify/dawn/pull/1097#issuecomment-1008946453 

cc @melissaperreault
